src/components/sidebar/SafeListContextMenu/MultiAccountContextMenu.tsx

```diff
@@ -0,0 +1,107 @@
+import type { MouseEvent } from 'react'
+import { useState, type ReactElement } from 'react'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import IconButton from '@mui/material/IconButton'
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import MenuItem from '@mui/material/MenuItem'
+import ListItemText from '@mui/material/ListItemText'
+
+import EntryDialog from '@/components/address-book/EntryDialog'
+import EditIcon from '@/public/images/common/edit.svg'
+import PlusIcon from '@/public/images/common/plus.svg'
+import ContextMenu from '@/components/common/ContextMenu'
+import { trackEvent, OVERVIEW_EVENTS, OVERVIEW_LABELS } from '@/services/analytics'
+import { SvgIcon } from '@mui/material'
+import { AppRoutes } from '@/config/routes'
+import router from 'next/router'
+import { CreateSafeOnNewChain } from '@/features/multichain/components/CreateSafeOnNewChain'
+
+enum ModalType {
+  RENAME = 'rename',
+  ADD_CHAIN = 'add_chain',
+}
+
+const defaultOpen = { [ModalType.RENAME]: false, [ModalType.ADD_CHAIN]: false }
+
+const MultiAccountContextMenu = ({
```

Comment:
about the architecture of this component, since it's a button that shows a dropdown with some options which triggers a modal, could we separate the dropdown in generic renderProps component which provides the `openMenu` function to the rendered child and takes an array of options? something like: 

```jsx
<DropdownMenu items={items}>
{(openMenu) =>  (
<IconButton data-testid="safe-options-btn" edge="end" size="small" onClick={handleOpenContextMenu}>
      <MoreVertIcon sx={({ palette }) => ({ color: palette.border.main })} />
</IconButton>
)}
</DropdownMenu>
```

in this case, `items` would be a common type like this:

```typescript
type item = {
 onClick: (e: MouseEvent) => void,
 label: string
 icon?: React.ReactNode
}[]
```

Reply:
The architecture also comes from the MUI component library. I do not think we can easily rewrite it to this. 
So I'll stick to how the examples use the component for the time being.